NewBlackAuthenticator/Controllers/TwoFARegistrationsController.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using NewBlackAuthenticator.Data;
+using NewBlackAuthenticator.Models;
+
+namespace NewBlackAuthenticator.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TwoFARegistrationsController : ControllerBase
+    {
+        private readonly NBADBcontext _context;
+
+        public TwoFARegistrationsController(NBADBcontext context)
+        {
+            _context = context;
+        }
+        private readonly static Random random = new Random();
```

Comment:
Nooooes! Do not use the `Random` class if you want to generate legitimately random tokens. The class you use here produces predictable output and is a strict no-no in security!

Check out the `RandomNumberGenerator` (newer) or `RNGCryptoServiceProvider` (EVA still uses this one - BUT NOT FOR LONG) instead.